react-example/cypress/integration/users/users.spec.ts

```diff
@@ -0,0 +1,155 @@
+describe('Update User', () => {
+  it('should paint 200 response data to screen', () => {
+    cy.intercept(
+      {
+        method: 'POST',
+        url: '/api/users/update*'
+      },
+      { fixture: 'users/200.json' }
+    ).as('updateUser');
+
+    cy.visit('/users');
+
+    cy.get('[data-qa-update-user-input]').type('SomeItem');
+    cy.get('[data-qa-update-user-submit]').submit();
+
+    cy.wait('@updateUser');
+
+    cy.get('[data-qa-update-user-response]').contains(
+      JSON.stringify({
+        msg: 'success mocked from cypress',
+        code: 'Success',
+        params: null
+      })
+    );
+  });
+
+  it('should paint 400 response data to screen', () => {
+    cy.intercept(
+      {
+        method: 'POST',
+        url: '/api/users/update*'
+      },
+      { fixture: 'users/400.json' }
+    ).as('updateUser');
+
+    cy.visit('/users');
+
+    cy.get('[data-qa-update-user-input]').type('SomeItem');
+    cy.get('[data-qa-update-user-submit]').submit();
+
+    cy.wait('@updateUser');
+
+    cy.get('[data-qa-update-user-response]').contains(
+      JSON.stringify({
+        msg: 'No API key found on request',
+        code: 'BadApiKey',
+        params: { ip: '68.111.205.112', endpoint: '/api/users/update' }
+      })
+    );
+  });
+});
+
+describe('Update User Email', () => {
+  it('should paint 200 response data to screen', () => {
+    cy.intercept(
+      {
+        method: 'POST',
+        url: '/api/users/updateEmail*'
+      },
+      { fixture: 'users/200.json' }
+    ).as('updateUserEmail');
+
+    cy.visit('/users');
+
+    cy.get('[data-qa-update-user-email-input]').type('SomeItem');
+    cy.get('[data-qa-update-user-email-submit]').submit();
+
+    cy.wait('@updateUserEmail');
+
+    cy.get('[data-qa-update-user-email-response]').contains(
+      JSON.stringify({
+        msg: 'success mocked from cypress',
+        code: 'Success',
+        params: null
+      })
+    );
+  });
+
+  it('should paint 400 response data to screen', () => {
+    cy.intercept(
+      {
+        method: 'POST',
+        url: '/api/users/updateEmail*'
+      },
+      { fixture: 'users/400.json' }
+    ).as('updateUserEmail');
+
+    cy.visit('/users');
+
+    cy.get('[data-qa-update-user-email-input]').type('SomeItem');
+    cy.get('[data-qa-update-user-email-submit]').submit();
+
+    cy.wait('@updateUserEmail');
+
+    cy.get('[data-qa-update-user-email-response]').contains(
+      JSON.stringify({
+        msg: 'No API key found on request',
+        code: 'BadApiKey',
+        params: { ip: '68.111.205.112', endpoint: '/api/users/update' }
```

Comment:
just double checking, is this hardcoded IP address meant to be here?

Reply:
it is, but good call! Let me make the error more generic